### PR TITLE
Sol record

### PR DESCRIPTION
**Pull request by dr497**

### CODE REVIEW

File: sns-book/src/domain-name/records.md [hunk 1]
```diff
@@ -72,13 +76,22 @@ A more generic resolving function `getRecord` is also exported with the followin
 (connection: Connection, domain: string, record: Record) => Promise<NameRegistryState>
 ```
 
+## The SOL record
+
+The SOL record can be used to receive funds to a different address than the one owning the domain. This allows people to hold the domain on a cold wallet while still being able to receive funds in a hot wallet.
```
Comment (ellttBen) [inline sns-book/src/domain-name/records.md]: while being able to receive funds _on_ a hot wallet

File: sns-book/src/domain-name/records.md [hunk 2]
```diff
@@ -72,13 +76,22 @@ A more generic resolving function `getRecord` is also exported with the followin
...
+
+The SOL record data contains a 96-byte array that is the concatenation of a public key (32 bytes) and signature (64 bytes). The first 32 bytes represent the public key (`pubkey`) to which funds should be sent and the next 64 bytes are the signature of `pubkey_as_bytes + record_key_as_bytes` signed by the owner of the domain. If the signature is invalid funds **must not** be transfered.
```
Comment (ellttBen) [inline sns-book/src/domain-name/records.md]: are the signature of `pubkey_as_bytes + record_key_as_bytes` by the owner of the domain

File: sns-book/src/domain-name/records.md [hunk 3]
```diff
@@ -72,13 +76,22 @@ A more generic resolving function `getRecord` is also exported with the followin
...
+
+The signature is required to prevent funds being sent to a stale SOL record after a domain was transfered or sold to a new owner.
```
Comment (ellttBen) [inline sns-book/src/domain-name/records.md]: _has been_ transferred or sold

File: sns-book/src/wallet-guide.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+![wallet-guide](./assets/wallet-guide.png)
+
+1. Check if the domain name is tokenized
+
+   - **Yes:** The correct destination of funds is the token holder
+   - **No:** Go to step 2
+
+2. Check the SOL record
+
+   - **If** the SOL record is set and the signature is valid, the public key specified in the record is the correct destination.
```
Comment (ellttBen) [inline sns-book/src/wallet-guide.md]: What happens if the signature is invalid? I don't think it's clear here

File: sns-book/src/wallet-guide.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+};
+```
+
+> Not resolving domains properly might lead to loss of funds ⚠️
+
+## FAQ
+
+1. What happens if funds are sent to the `NameRegistry` owner when the domain is tokenized?
+
+As long as the user owns the tokenized domains (i.e the NFT) they will be able to withdraw from the PDA escrow that received the funds. However, if for some reason the user does not own the NFT he won't be able to withdraw the funds.
```
Comment (ellttBen) [inline sns-book/src/wallet-guide.md]: _they_ won't be able to withdraw the funds

File: sns-book/src/wallet-guide.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+2. Why is there a signature in the SOL record?
+
+The SOL record data contains a 96-byte array that is the concatenation of a public key (32 bytes) and signature (64 bytes). The first 32 bytes represent the public key (`pubkey`) to which funds should be sent and the next 64 bytes are the signature of `pubkey_as_bytes + record_key_as_bytes` signed by the owner of the domain. If the signature is invalid funds **must not** be transfered.
+
+The signature is required to prevent funds being sent to a stale SOL record after a domain was transfered or sold to a new owner.
```
Comment (ellttBen) [inline sns-book/src/wallet-guide.md]: a domain _has been_ transferred or sold

File: sns-book/src/wallet-guide.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+## FAQ
+
+1. What happens if funds are sent to the `NameRegistry` owner when the domain is tokenized?
+
+As long as the user owns the tokenized domains (i.e the NFT) they will be able to withdraw from the PDA escrow that received the funds. However, if for some reason the user does not own the NFT they won't be able to withdraw the funds.
+
+2. Why is there a signature in the SOL record?
+
+The SOL record data contains a 96-byte array that is the concatenation of a public key (32 bytes) and signature (64 bytes). The first 32 bytes represent the public key (`pubkey`) to which funds should be sent and the next 64 bytes are the signature of `pubkey_as_bytes + record_key_as_bytes` signed by the owner of the domain. If the signature is invalid funds **must not** be transfered.
```
Comment (Lcchy) [inline sns-book/src/wallet-guide.md]: transferred

**Pull request merged**